xrpVendordep/src/main/native/cpp/xrp/XRPGyro.cpp

```diff
@@ -26,70 +27,70 @@ XRPGyro::XRPGyro() : m_simDevice("Gyro:XRPGyro") {
   }
 }
 
-double XRPGyro::GetAngle() const {
+units::radian_t XRPGyro::GetAngle() const {
   return GetAngleZ();
 }
 
 frc::Rotation2d XRPGyro::GetRotation2d() const {
-  return frc::Rotation2d{units::degree_t{GetAngle()}};
+  return frc::Rotation2d{GetAngle()};
 }
 
-double XRPGyro::GetRate() const {
+units::radians_per_second_t XRPGyro::GetRate() const {
   return GetRateZ();
 }
 
-double XRPGyro::GetRateX() const {
+units::radians_per_second_t XRPGyro::GetRateX() const {
   if (m_simRateX) {
-    return m_simRateX.Get();
+    return units::degrees_per_second_t{m_simRateX.Get()};
   }
 
-  return 0.0;
+  return 0.0_rad_per_s;
 }
 
-double XRPGyro::GetRateY() const {
+units::radians_per_second_t XRPGyro::GetRateY() const {
   if (m_simRateY) {
-    return m_simRateY.Get();
+    return units::degrees_per_second_t{m_simRateY.Get()};
   }
 
-  return 0.0;
+  return 0.0_rad_per_s;
```

Comment:
The `.0` suffix is redundant for these user-defined literals.
```suggestion
  return 0_rad_per_s;
```